designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/UsageStatisticsReportsSettingsDeterminer.scala

```diff
@@ -46,20 +54,72 @@ object UsageStatisticsReportsSettingsDeterminer {
       processService
         .getLatestProcessesWithDetails(
           ScenarioQuery.unarchived,
-          GetScenarioWithDetailsOptions.detailsOnly.withFetchState
+          GetScenarioWithDetailsOptions.withsScenarioGraph.withFetchState
         )(user)
         .map { scenariosDetails =>
           scenariosDetails.map(scenario =>
             ScenarioStatisticsInputData(
               scenario.isFragment,
               scenario.processingMode,
               deploymentManagerTypeByProcessingType(scenario.processingType),
-              scenario.state.map(_.status)
+              scenario.state.map(_.status),
+              scenario.scenarioGraph.map(_.nodes.length).getOrElse(0),
+              scenario.processCategory,
+              scenario.processVersionId,
+              scenario.createdBy,
+              scenario.scenarioGraph
+                .map { graph =>
+                  graph.nodes.map {
+                    case _: FragmentInput => 1
+                    case _                => 0
+                  }.sum
+                }
+                .getOrElse(0)
             )
           )
         }
     }
-    new UsageStatisticsReportsSettingsDeterminer(config, fingerprintSupplier, fetchNonArchivedScenarioParameters)
+    def fetchActivity(): Future[List[DbProcessActivityRepository.ProcessActivity]] = {
+      implicit val user: LoggedUser = NussknackerInternalUser.instance
+      for {
+        scenarioDetails <- processService.getLatestProcessesWithDetails(
+          ScenarioQuery.unarchived,
+          GetScenarioWithDetailsOptions.detailsOnly.withFetchState
+        )
+        activity <- scenarioDetails.map { scenario =>
+          scenarioActivityRepository.findActivity(scenario.processIdUnsafe)
+        }.sequence
+      } yield activity
+    }
+
+    def fetchComponentList(): Future[List[ComponentListElement]] = {
+      implicit val user: LoggedUser = NussknackerInternalUser.instance
+      componentService.getComponentsList
+    }
+
+    def fetchScenarioActions(): Future[List[List[ProcessAction]]] = {
```

Comment:
Shouldn't it be: `Future[List[ProcessAction]]`?

Reply:
I think of it as List of `ProcessAction` per scenario, hence List of Lists, but I may be wrong in my understanding

Reply:
Now it's `Future[List[ProcessSaction]]` as we gather statistics only for currently running scenarios and so we only need `lastDeployedAction`